networkx/algorithms/centrality/domirank.py

```diff
@@ -0,0 +1,303 @@
+    :func:`~networkx.algorithms.link_analysis.pagerank_alg.pagerank`
+    :func:`~networkx.algorithms.link_analysis.hits_alg.hits`
+
+    References
+    ----------
+    .. [1] Engsig, M., Tejedor, A., Moreno, Y. et al.
+        "DomiRank Centrality reveals structural fragility of complex networks via node dominance."
+        Nat Commun 15, 56 (2024). https://doi.org/10.1038/s41467-023-44257-0
+    """
+    import numpy as np
+    import scipy as sp
+
+    if len(G) == 0:
+        raise nx.NetworkXPointlessConcept(
+            "cannot compute centrality for the null graph"
+        )
+    GAdj = nx.to_scipy_sparse_array(G)  # convert to scipy sparse csr array
+
+    # Here we renormalize alpha with the smallest eigenvalue (most negative eigenvalue) by calling the "hidden" function _find_smallest_eigenvalue()
+    # Note, this function always uses the recursive definition
+    if analytical == False:
+        # If the recursive formulation is used, the alpha has to be bounded (competition parameter).
+        if alpha < 0:
+            raise nx.NetworkXUnfeasible(
+                "the competition parameter alpha must be positive and bounded such that alpha: [0,1]"
+            )
+        if alpha > 1:
+            raise nx.NetworkXUnfeasible(
+                "supercharging the competition parameter (alpha > 1) requires the <analytical = True> flag"
+            )
+        sigma = np.abs(
+            alpha
+            / _find_smallest_eigenvalue(
+                GAdj,
+                maxDepth=int(max_iter / 5),
+                dt=dt,
+                epsilon=epsilon,
+                max_iter=max_iter,
+                patience=patience,
+            )
+        )
+        # store this to prevent more redundant calculations in the future
+        pGAdj = sigma * GAdj.astype(np.float32)
+        # initalize a proportionally (to system size) small non-zero uniform vector
+        Psi = np.ones(pGAdj.shape[0]).astype(np.float32) / pGAdj.shape[0]
+        # initialize a zero array to store values (yes, this could be done with a smaller array with some smart indexing, but this is not computationally or memory heavy)
+        maxVals = np.zeros(int(max_iter / patience)).astype(np.float32)
+        # ensure dt is a float
+        dt = np.float32(dt)
+        # start a counter
+        j = 0
+        # set up a boundary condition for stopping divergence
+        boundary = epsilon * pGAdj.shape[0] * dt
+        for i in range(max_iter):
+            # DomiRank recursive definition
+            tempVal = ((pGAdj @ (1 - Psi)) - Psi) * dt
+            # Newton iteration addition step
+            Psi += tempVal.real
+            # Here we do the checking to see if we are diverging
+            if i % patience == 0:
+                if np.abs(tempVal).sum() < boundary:
+                    break
+                maxVals[j] = tempVal.max()
+                if i == 0:
+                    initialChange = maxVals[j]
```

Comment:
I don't think this value ever gets used, so maybe you can get rid of the `if i == 0` block?